app/services/courses_query.rb

```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+class CoursesQuery
+  def self.call(params = {})
+    new(params:).call
+  end
+
+  attr_reader :scope
+
+  def initialize(params:)
+    @params = params
+    @scope = RecruitmentCycle
+             .current
+             .courses
+             .joins(:site_statuses)
+             .where(
+               site_statuses: {
+                 status: SiteStatus.statuses[:running],
+                 publish: SiteStatus.publishes[:published]
+               }
+             )
```

Comment:
One of the rules @Nitemaeric and I talked was to decrease the usage of scopes so moved the where here for the `findable` scope. This is easier to read and reduce the too much indirection when reading code.

Reply:
This is building on tech debt around the "running" of `SiteStatus`. We are supposed to be moving away from running sites and toward open / closed courses. The class should just be called `CourseSite` like the table now.

Where is `published_on_ucas` defined? I can't find it anywhere :thinking:

Reply:
> This is building on tech debt around the "running" of SiteStatus. We are supposed to be moving away from running sites and toward open / closed courses.

Interesting. So the current scope `Course.findable` are using like that. So should we move away from findable scope and make open / closed? We do have the filter for open and closed applications to be added in a new ticket.

> Where is published_on_ucas defined? I can't find it anywhere 🤔

It is an enumerator.
```ruby
# site status
  31   enum :publish, {
  32     published: 'Y',
  33     unpublished: 'N'
  34   }, suffix: :on_ucas
```

Reply:
Iain and I discussed this, I will create a card specifically to improve to have a course status. 

Would be awesome if we could do:

```ruby
     @scope = RecruitmentCycle
              .current
             .courses
             .where(status: 'published')
```

But we can't at the moment.

Whenever we change the course enrichment  to "publish" a course (or changes to withdrawn) we fill the status of "published" or "withdrawn".

Reply:
So as you mentioned to me in a huddle, we need to be able to tell if the course is published or not.
Ideally we'd have a column 'status' or similar. Instead we have CourseEnrichment where the `status` is stored inside a jsonb object.

Happy to approve this now that we've discussed how we can improve it in future.